### ISSUE TITLE
Do all non-system versions get removed with every brew upgrade?

### ISSUE THREAD

**Issue by idmit** [pyenv/pyenv#815]
Sometimes all python versions that are installed (except for the system one) disappear. This time I noticed that I did a `brew upgrade` fairly recently.

Does homebrew erase everything on each upgrade and it is the expected behaviour?

**Comment by idmit** [comment]
Oh, I think I might have misled you, closing this.

**Comment by idmit** [comment]
No, it happened again. What might be the problem?

**Comment by naglalakk** [comment]
Hi I think this is happening to me as well. I use python 3.5.2 on multiple projects, I've already setup a virtualenv for them and this has worked, installed dependencies etc. I did a brew upgrade yesterday and now when doing pyenv versions it says 'version `3.5.2' is not installed'. Suddenly python2.7 is my default version.

**Comment by idmit** [comment]
I think, the problem is that pyenv stores installed versions under `./versions` relative path. When brew upgrades, it completely replaces pyenv directory, which means it removes the contents of `./versions` too.

I've explicitly set `PYENV_ROOT` variable to `~/.pyenv`, in order to store everything under `~/.pyenv/versions` now. I then tried reinstalling pyenv and everything seems to work properly this time. Just make sure to export `PYENV_ROOT` before calling `pyenv init`. I also manually expanded `~` to `/Users/usrname` just to be safe.

@naglalakk hope this helps!
